packages/test/test-end-to-end-tests/src/test/deRehydrateContainerTests.spec.ts

```diff
@@ -311,6 +311,44 @@ describeFullCompat(`Dehydrate Rehydrate Container Test`, (getTestObjectProvider)
             assert.strictEqual(sparseMatrix.id, sparseMatrixId, "Sparse matrix should exist!!");
         });
 
+        it("Rehydrate container multiple times round trip serialize/deserialize", async () => {
+            const { container } =
+                await createDetachedContainerAndGetRootDataStore();
+            const snapshotTree1 = container.serialize();
+            const container1 = await loader.rehydrateDetachedContainerFromSnapshot(snapshotTree1);
+
+            const snapshotTree2 = container1.serialize();
+            const container2 = await loader.rehydrateDetachedContainerFromSnapshot(snapshotTree2);
```

Comment:
As long as we're doing it twice, maybe we could do it a bunch of times in a loop? It should work for any number of serialization/rehydrations, right?

Reply:
Yes. That was the issue which was causing it to do so.

Reply:
Adding a loop.